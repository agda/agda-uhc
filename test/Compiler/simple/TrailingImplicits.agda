module TrailingImplicits where

-- see also https://lists.chalmers.se/pipermail/agda-dev/2015-January/000041.html

open import Common.IO
open import Common.Unit
open import Common.Nat

f : (m : Nat) {l : Nat} -> Nat
f zero {l = l} = l
f (suc y) = y

main : IO Unit
main = printNat (f 0 {1}) ,,
  putStr "\n" ,,
  printNat (f 30 {1})
