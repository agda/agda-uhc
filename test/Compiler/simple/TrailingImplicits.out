EXECUTED_PROGRAM

ret > ExitSuccess
out > 1
out > 29
